src/AspNet.Security.OAuth.Validation/Events/IOAuthValidationEvents.cs

```diff
@@ -0,0 +1,22 @@
+﻿using System.Threading.Tasks;
+
+namespace AspNet.Security.OAuth.Validation
+{
+    public interface IOAuthValidationEvents
+    {
+        /// <summary>
+        /// Invoked when an access token is first received.
+        /// </summary>
+        Task AccessTokenReceived(AccessTokenReceivedContext context);
+
+        /// <summary>
+        /// Invoked after processing, when a token has been validated.
+        /// </summary>
+        Task TokenValidated(TokenValidatedContext context);
+
+        /// <summary>
+        /// Invoked when audiences are to be validated for a message.
+        /// </summary>
+        Task ValidateAudience(ValidateAudienceContext context);
```

Comment:
Yeah, I was focusing on one middleware at a time and ran into the issue I had. It'll be gone :+1: